deploy/terraform/modules/app_tier/infrastructure.tf

```
@@ -132,9 +149,9 @@ resource "azurerm_lb" "web" {
 
   frontend_ip_configuration {
     name                          = "sap${lower(local.application_sid)}web"
-    subnet_id                     = local.sub_app_exists ? data.azurerm_subnet.subnet-sap-app[0].id : azurerm_subnet.subnet-sap-app[0].id
+    subnet_id                     = local.sub_web_defined ? local.sub_web_exists ? data.azurerm_subnet.subnet-sap-web[0].id : azurerm_subnet.subnet-sap-web[0].id : local.sub_app_exists ? data.azurerm_subnet.subnet-sap-app[0].id : azurerm_subnet.subnet-sap-app[0].id
     private_ip_address_allocation = "Static"
-    private_ip_address            = cidrhost(local.sub_app_prefix, local.ip_offsets.web_lb)
+    private_ip_address            = local.sub_web_defined ? cidrhost(local.sub_web_prefix, local.ip_offsets.web_lb) : cidrhost(local.sub_app_prefix, local.ip_offsets.web_lb)
```

This way it is shorter:
```suggestion
    private_ip_address            = cidrhost( local.sub_web_defined ? local.sub_web_prefix : local.sub_app_prefix, local.ip_offsets.web_lb )
```